File: Leetcode/1325 Delete Leaves With a Given Value.ts
function removeLeafNodes(root: TreeNode | null, target: number): TreeNode | null {
    if (isLeafWithTarget(root, target)) {
        return null;
    }
    root.left = removeLeafNodes(root.left, target);
    root.right = removeLeafNodes(root.right, target);
    return isLeafWithTarget(root, target) ? null : root;
}

function isLeafWithTarget(root: TreeNode | null, target: number) {
    return root === null || (root.left === null && root.right === null && root.val === target);
}

File: Leetcode/1325 Delete Leaves With a Given Value.py
class Solution:
    def removeLeafNodes(self, root, target):
        if is_leaf_with_target(root, target):
            return None
        root.left = self.removeLeafNodes(root.left, target)
        root.right = self.removeLeafNodes(root.right, target)
        return None if is_leaf_with_target(root, target) else root

def is_leaf_with_target(root, target):
    return root is None or (root.left is None and root.right is None and root.val == target)
